src/main/webapp/js/userMap.js

```diff
@@ -104,8 +114,98 @@ function handleData(err, userData) {
             }
         }
     });
+
+    map.addPlugin('pins', function ( layer, data, options ) {
+        var self = this,
+            fillData = this.options.fills,
+            svg = this.svg;
+
+        if ( !data || (data && !data.slice) ) {
+            throw "Datamaps Error - markers must be an array";
+        }
+
+        var markers = layer.selectAll('image.datamaps-pins').data( data, JSON.stringify );
+
+        markers
+        .enter()
+        .append('image')
+        .attr('class', 'datamaps-pin')
+        .attr('xlink:href', 'images/pin.png')
+        .attr('height', 20)
+        .attr('width', 20)
+        .attr('x', function (datum) {
+            var latLng;
+            if (datumHasCoords(datum)) {
+                latLng = self.latLngToXY(datum.latitude, datum.longitude);
+            } else if ( datum.centered ) {
+                latLng = self.path.centroid(svg.select('path.' + datum.centered).data()[0]);
+            }
+            if ( latLng ) {
+                return latLng[0];
+            }
+        })
+        .attr('y', function ( datum ) {
+            var latLng;
+            if (datumHasCoords(datum)) {
+                latLng = self.latLngToXY(datum.latitude, datum.longitude);
+            } else if (datum.centered) {
+                latLng = self.path.centroid(svg.select('path.' + datum.centered).data()[0]);
+            }
+            if (latLng) {
+                return latLng[1];
+            }
+        })
+        .on('mouseover', function ( datum ) {
+            var $this = d3.select(this);
+
+            if (options.popupOnHover) {
+                self.updatePopup($this, datum, options, svg);
+            }
+        })
+        .on('mouseout', function ( datum ) {
+            var $this = d3.select(this);
+
+            if (options.highlightOnHover) {
+                var previousAttributes = JSON.parse( $this.attr('data-previousAttributes') );
+                for (var attr in previousAttributes) {
+                    $this.style(attr, previousAttributes[attr]);
+                }
+            }
+            d3.selectAll('.datamaps-hoverover').style('display', 'none');
+        });
+
+        markers.exit()
+        .transition()
+        .delay(options.exitDelay)
+        .attr("height", 0)
+        .remove();
+
+        function datumHasCoords (datum) {
+            return typeof datum !== 'undefined' && typeof datum.latitude !== 'undefined' && typeof datum.longitude !== 'undefined';
+        }
+
+    });
+
+    map.pins(pins, {
+        popupOnHover: true,
+        popupTemplate: function(data, b) {
+          return '<div class="hoverinfo">'
+                     + '<p>'
+                         + '<span class="bold">'
+                         + data.name
+                         + '</span>'
+                         + '<br>'
+                         + 'Institution(s): '
+                         + data.numOfInstitutions
+                     + '</p>'
+                 + '</div>';
```

Comment:
There is a similar usage of this return value before. See if you can extract it out to a method so that both usages can simply call that method.